application_test.go

```diff
@@ -294,3 +294,17 @@ func TestApplication(t *testing.T) {
 	_, ok = err.(*APIError)
 	assert.False(t, ok)
 }
+
+func TestApplicationConfiguration(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+
+	application, err := endpoint.Client.ApplicationConfiguration(fakeAppName, "2014-09-12T23:28:21.737Z")
+	assert.NoError(t, err)
+	assert.NotNil(t, application)
+	assert.Equal(t, application.ID, fakeAppName)
+	assert.NotNil(t, application.HealthChecks)
+	assert.NotNil(t, application.Tasks)
+	assert.Equal(t, len(application.HealthChecks), 1)
+	assert.Equal(t, len(application.Tasks), 2)
+}
```

Comment:
We should also test the unhappy path quickly by either one of two approaches:

1. Call the new method with with invalid name and verify that an error is returned (basically copying-and-pasting the very logic that [TestApplication() already has](https://github.com/gambol99/go-marathon/blob/master/application_test.go#L281-L282)).
1. Extract the test assertion logic common to `TestApplication` and your new test function into a non-exported helper function and call it from each test.

I prefer no. 2: While code redundancy isn't necessarily a bad thing in unit-testing, here I think it's actually desirable to get rid of it because we want all configuration-retrieving methods to return the same content.